File: 0110-balanced-binary-tree/0110-balanced-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    boolean f = true;
    public boolean isBalanced(TreeNode root) {
        helper(root);
        return f;
    }
    public int helper(TreeNode root){
        if(root == null) return 0;
        int left = helper(root.left) + 1;
        int right = helper(root.right) + 1;
        if(Math.abs(left - right) > 1) f = false;
        return Math.max(left , right);
    }
}

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        self.flag = True
        self.helper(root)
        return self.flag
    
    def helper(self, root: Optional[TreeNode]) -> bool:
        if not root:
            return 0
        left, right = self.helper(root.left) + 1, self.helper(root.right) + 1
        if abs(left - right) > 1:
            self.flag = False
        return max(left, right)